packages/router/src/lib/helpers/shared.ts

```diff
@@ -61,12 +111,21 @@ export const calculateGasFeeInReceivingToken = async (
   if (chaindIdsForGasFee.includes(receivingChainId)) {
     const gasLimitForPrepare = BigNumber.from(GAS_ESTIMATES.prepare);
     const [ethPriceInReceivingChain, receivingTokenPrice, gasPriceInReceivingChain] = await Promise.all([
-      getTokenPrice(receivingChainId, constants.AddressZero, requestContext),
-      getTokenPrice(receivingChainId, receivingAssetId, requestContext),
+      getTokenPrice(tokenPricingReceivingChain, constants.AddressZero, requestContext),
+      getTokenPrice(tokenPricingReceivingChain, tokenPricingAssetIdReceivingChain, requestContext),
       getGasPrice(receivingChainId, requestContext),
     ]);
 
-    const gasAmountInUsd = gasPriceInReceivingChain.mul(gasLimitForPrepare).mul(ethPriceInReceivingChain);
+    // https://community.optimism.io/docs/users/fees-2.0.html#fees-in-a-nutshell
```

Comment:
with the hardcoded comment above, this comment could go above that constant at the top of the file

Reply:
this is a different comment -- this is for adding l1 fees that is *separate* from the oracle concerns